src/main/java/org/jenkinsci/plugins/github/pullrequest/builders/GitHubPRStatusBuilder.java

```diff
@@ -52,41 +56,44 @@ public GitHubPRMessage getStatusMessage() {
     }
 
     @Override
-    public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener)
-            throws InterruptedException, IOException {
-        GitHubPRTrigger trigger = build.getProject().getTrigger(GitHubPRTrigger.class);
+    public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath workspace, @Nonnull Launcher launcher,
+                        @Nonnull TaskListener listener) throws InterruptedException, IOException {
+        // No triggers in Run class, but we need it
+        final GitHubPRTrigger trigger = JobInfoHelpers.triggerFrom(run.getParent(), GitHubPRTrigger.class);
         if (trigger == null) {
-            return true;
+            // silently skip. TODO implement error handler, like in publishers
+            return;
         }
 
-        GitHubPRCause cause = build.getCause(GitHubPRCause.class);
+        GitHubPRCause cause = run.getCause(GitHubPRCause.class);
         if (cause == null) {
-            return true;
+            return;
```

Comment:
perform for `Run`s doesn't expect `boolean`. And boolean as return value is old style, it was replaced with `AbortException`. There are 2 ways of handling steps misuses (configured step but has no expected data):
- fail 
- silently skip.
So this code do only silent skip (for publishers i wrote errorHandler with UI).